examples/sol-contract/src/processor.rs

```diff
@@ -0,0 +1,110 @@
+//! Program instruction processor
+//! Only the program admin can issue the Init instruction.
+//! And anyone can check the loan with the Loan2Value instruction.
+
+use solana_program::account_info::{
+    next_account_info,
+    AccountInfo,
+};
+use solana_program::entrypoint::ProgramResult;
+use solana_program::msg;
+use solana_program::program_error::ProgramError;
+use solana_program::program_pack::{
+    IsInitialized,
+    Pack,
+};
+use solana_program::pubkey::Pubkey;
+
+use borsh::BorshDeserialize;
+use pyth_sdk_solana::load_price_feed_from_account_info;
+
+use crate::instruction::ExampleInstructions;
+use crate::state::AdminConfig;
+
+pub fn process_instruction(
+    _program_id: &Pubkey,
+    _accounts: &[AccountInfo],
+    input: &[u8],
+) -> ProgramResult {
+    let account_iter = &mut _accounts.iter();
+    let signer = next_account_info(account_iter)?;
+    let data_account = next_account_info(account_iter)?;
+    let pyth_loan_account = next_account_info(account_iter)?;
+    let pyth_collateral_account = next_account_info(account_iter)?;
+
+    let instruction = ExampleInstructions::try_from_slice(input)?;
+    match instruction {
+        ExampleInstructions::Init {} => {
+            if !(signer.key == _program_id && signer.is_signer) {
```

Comment:
I've never seen this usage of the program authority. It's good because only the deployer has the key. @jayantk is it a good practice? What's a good alternative?

Reply:
i think this is fine for the example. people will have different permissioning systems depending on their use case. I suggest simply leaving a code comment like “Only an authorized / trusted key should be able to configure the price feed id  for each asset”